[PCF] Add AF in list of allowed NF types for PCF

herlesupreeth opened the pull request:
Added AF type in allowed NF types when PCF advertises to NRF

acetcom commented:
@herlesupreeth 

Thanks for the PR adding the AF type to the allowed NF types when PCF advertises to the NRF. I ran the /tests/registration/registration simple-test and saw the following error:

```
05/04 09:06:12.448: [sbi] ERROR: Not allowed nf-type[PCF] in nf-service[npcf-policyauthorization] (../lib/sbi/context.c:1562)
05/04 09:06:12.448: [pcf] FATAL: pcf_nbsf_management_build_register: Assertion `nf_instance’ failed. (../src/pcf/nbsf-build.c:76)
05/04 09:06:12.448: [core] FATAL: backtrace() returned 14 addresses (../lib/core/ogs-abort.c:37)
```

I haven’t had a chance to dig into this yet. Please let me know if you need any more details or have questions.

Thanks a lot!
Sukchan

herlesupreeth commented:
Ah, my bad. I will take a look and fix them

herlesupreeth commented:
Basically in nbsf-build.c (in pcf) there is the following code which is trying to find the NF instance based on service name but the requester NF type is set to PCF. This was failing when I had added  `ogs_sbi_nf_service_add_allowed_nf_type(service, OpenAPI_nf_type_AF);` at line 72 of `sbi-path.c` (in pcf). So I removed this addition from the PR. Let me know if you need me to modify anything else.

```
requester_nf_type = NF_INSTANCE_TYPE(ogs_sbi_self()->nf_instance);
    ogs_assert(requester_nf_type);
    nf_instance = ogs_sbi_nf_instance_find_by_service_type(
                    OGS_SBI_SERVICE_TYPE_NPCF_POLICYAUTHORIZATION,
                    requester_nf_type);
```

herlesupreeth commented:
Thank you for the patch. I wasn't sure adding PCF to allowed NF types (for PCF authorization) made sense that's why I was hesitant. I will do the changes as suggested :)
